gammapy/estimators/energydependence.py

```diff
@@ -0,0 +1,275 @@
+from .core import Estimator
+
+__all__ = ["weighted_chi2_parameter", "EnergyDependenceEstimator"]
+
+
+def weighted_chi2_parameter(results_edep, parameter="sigma"):
+    """Calculate the weighted chi-squared value for the parameter of interest.
+
+    Parameters
+    ----------
+    result_edep : dict
+        Dictionary of results for the energy-dependent estimator.
+    parameter : str, optional
+        The model parameter to calculate the chi-squared value for.
+        Default is "sigma".
+
+    Returns
+    -------
+    chi2_result : dict
+        Dictionary with the chi-squared value for parameter of interest.
+    """
+
+    table_edep = Table(results_edep)
+
+    values = table_edep[parameter][1:]
+    errors = table_edep[f"{parameter}_err"][1:]
+
+    weights = 1 / errors**2
+    avg = np.average(values, weights=weights)
+
+    chi2_value = np.sum((values - avg) ** 2 / errors**2)
+    df = len(values) - 1
+    sigma_value = ts_to_sigma(chi2_value, df)
+
+    chi2_result = {}
+    chi2_result[f"chi2 {parameter}"] = [chi2_value]
+    chi2_result["df"] = [df]
+    chi2_result["significance"] = [sigma_value]
+
+    return chi2_result
+
+
+class EnergyDependenceEstimator(Estimator):
+    """Test if there is any energy-dependent morphology in a map dataset for a given set of energy bins.
+
+    Parameters
+    ----------
+    energy_edges : `~astropy.units.Quantity`
+        Energy edges for the energy-dependence test.
+    source : str or int
+        For which source in the model to compute the estimator.
+    fit : `~gammapy.modeling.Fit`, optional
+        Fit instance specifying the backend and fit options.
+        If None, the fit backend default is minuit.
+        Default is None.
+
+    """
+
+    tag = "EnergyDependenceEstimator"
+
+    def __init__(self, energy_edges, source=0, fit=None):
+
+        self.energy_edges = energy_edges
+        self.source = source
+        self.num_energy_bands = len(self.energy_edges) - 1
+
+        if fit is None:
+            fit = Fit(optimize_opts={"print_level": 1})
+
+        self.fit = fit
+
+    def _slice_datasets(self, datasets):
+        """Calculate a dataset for each energy slice.
+
+        Parameters
+        ----------
+        datasets : `~gammapy.datasets.Datasets`
+            Input datasets to use.
+
+        Returns
+        -------
+        slices_src : `~gammapy.datasets.Datasets`
+            Sliced datasets.
+        """
+        model = datasets.models[self.source]
+
+        slices_src = Datasets()
+        for i, (emin, emax) in enumerate(
+            zip(self.energy_edges[:-1], self.energy_edges[1:])
+        ):
+            for dataset in datasets:
+                sliced_src = dataset.slice_by_energy(
+                    emin, emax, name=f"{self.source}_{i}"
+                )
+                bkg_sliced_model = FoVBackgroundModel(dataset_name=sliced_src.name)
+                sliced_src.models = [
+                    model.copy(name=f"{sliced_src.name}"),
```

Comment:
This will remove the other `models` from the fit. Is that what you want? 
One might want to search for source extension in the presence of other sources (eg, diffuse emission)

Reply:
No, that is not the desired behaviour. 
I will fix this!